src/items.rs

```diff
@@ -1220,25 +1220,40 @@ impl Rewrite for ast::StructField {
         }
 
         let type_annotation_spacing = type_annotation_spacing(context.config);
-        let result = match name {
-            Some(name) => {
-                format!("{}{}{}{}:{}",
-                        attr_str,
-                        vis,
-                        name,
-                        type_annotation_spacing.0,
-                        type_annotation_spacing.1)
-            }
+        let mut result = match name {
+            Some(name) => format!("{}{}{}{}:", attr_str, vis, name, type_annotation_spacing.0),
             None => format!("{}{}", attr_str, vis),
         };
 
-        let last_line_width = last_line_width(&result);
+        let last_line_width = last_line_width(&result) + type_annotation_spacing.1.len();
         let budget = try_opt!(shape.width.checked_sub(last_line_width));
-        let rewrite = try_opt!(self.ty
-                                   .rewrite(context,
-                                            Shape::legacy(budget,
-                                                          shape.indent + last_line_width)));
-        Some(result + &rewrite)
+        let type_rewrite = self.ty
+            .rewrite(context,
+                     Shape::legacy(budget, shape.indent + last_line_width));
+        match type_rewrite {
+            Some(ty) => {
```

Comment:
@nrc Thank you for you feedbacks!
I have a question here, is it better to format in the following way?
In the first struct its field does not exceed 100 letters, but contains linebreaks.
```rust
struct Deep {
    deeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeep: node::Handle<IdRef<'id, Node<K, V>>,
                                                                        Type,
                                                                        NodeType>
```
to
```rust
struct Deep {
    deeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeep:
        node::Handle<IdRef<'id, Node<K, V>>, Type, NodeType>,
}
```